Update Doc-String of MapDriver.get_iter (#61)
merantix-momentum/squirrel-core: docs/usage/Iterstream.rst

.. code-block:: python

    def io_bound(item):
        print(f"{item} io_bound")
        time.sleep(1)
        return item

    it = IterableSource([1, 2, 3]).async_map(io_bound, max_workers=4).async_map(io_bound, max_workers=None)
    t1 = time.time()
    for i in it:
        print(i)
    print(time.time() - t1)

By default, :py:meth:`async_map <squirrel.iterstream.base.Composable.async_map>`
instantiates a :py:class:`ThreadPoolExecutor <concurrent.futures.ThreadPoolExecutor>` (`executor=None`).
It also accepts :py:class:`ProcessPoolExecutor <concurrent.futures.ProcessPoolExecutor>`,
which is a good choice when performing cpu-bound operations on a single machine.

The argument `max_workers` defines the maximum number of workers/threads the
:py:class:`ThreadPoolExecutor <concurrent.futures.ThreadPoolExecutor>`
uses when `exectuor=None`.
By default, `max_workers=None` relies on an internal heuristic of
the :py:class:`ThreadPoolExecutor <concurrent.futures.ThreadPoolExecutor>`
to select a reasonable upper bound.
This may differ between Python versions.
See the documentation of
:py:class:`ThreadPoolExecutor <concurrent.futures.ThreadPoolExecutor>` for details.

> In the above example, two :py:class:`ThreadPoolExecutor <concurrent.futures.ThreadPoolExecutor>`'s

Reviewer: ```suggestion
In the above example, two :py:class:`ThreadPoolExecutor <concurrent.futures.ThreadPoolExecutor>`\s
```

        time.sleep(1)
        return item

>     it = IterableSource([1, 2, 3]).async_map(io_bound, executor=tpool, max_workers=4).async_map(io_bound)

Reviewer: I suggest keeping `max_workers` here, removing `tpool` and explaining how to specify an `executor` below the code block (which you have already done)
```suggestion
    it = IterableSource([1, 2, 3]).async_map(io_bound, max_workers=4).async_map(io_bound)
```

Author: :+1: I will add this to the documentation

    t1 = time.time()
    for i in it:
        print(i)
    print(time.time() - t1)

:py:meth:`async_map <squirrel.iterstream.base.Composable.async_map>`
instantiates a :py:class:`ThreadPoolExecutor <concurrent.futures.ThreadPoolExecutor>`
if the argument `executor` is `None` (default).
It also accepts :py:class:`ProcessPoolExecutor <concurrent.futures.ProcessPoolExecutor>`,
which is a good choice when performing cpu-bound operations on a single machine.

> The argument `max_workers` defines the maximum number of workers of the selected
> `executor`. By default, `max_workers=None` relies on an internal heuristic of
> the `executor` to select a reasonable upper bound.

Reviewer: We should make explicit when exactly `max_workers` is being used.
```suggestion
If an executor is not provided, the argument `max_workers` can be used to control the level of parallelism.
This argument defines the maximum number of workers used by the ThreadPoolExecutor
that will be instantiated by squirrel.
By default, `max_workers=None` relies on an internal heuristic of
ThreadPoolExecutor to select a reasonable upper bound.
```

This may differ between Python versions.

> See the documentation of the `executor` for more details.

Reviewer: ```suggestion
See the documentation of the `executor` for more details.
If an `executor` is specified, then `max_workers` is ignored.
```
